### PR TITLE
FI new function for users

### LINKED ISSUE #1757: [FEAT] Enable explicit control of FI calculation
**Is your feature request related to a problem? Please describe.**
We want the user to be able to use our feature importance calculation mechanism easily in two scenarios:
1. Passing calculated feature importance to a check/suite 
2. Calculating feature importance and exporting the FI Series to an external source

**Describe the solution you'd like**
Currently if the user wants to control the FI calculation he can either pass a FI Series he calculated elsewhere, or modify one of the following parameters of the run() method:
```
        feature_importance_force_permutation: bool = False,
        feature_importance_timeout: int = 120,
```
We want the option for an intermediate flow - use something equivalent to our `calculate_feature_importance` method (perhaps with the permutation_kwargs explicitly stated), get an FI Series that can be inspected, and then pass it to the run() feature_importance arg.

Also add to docs

**Describe alternatives you've considered**
This creates a "two ways to do one thing" situation, so need to consider if there is a way to simplify.

*Additional alternative*: create a sub package with SHAP that can calculate way better FI compare to permutation importance.

### PR DESCRIPTION
**Pull request by nirhutnik**
Closes #1757 

Added function calculate_feature_importance in new file tabular/user_utils.py for users to use.
Changed the warnings for force_permutation=True so they will fail the process and we won't continue to built-in feature importance.
Added paragraph in feature_importance.rst.

NOTE!
Changed behavior of calculate_feature_importance in features.py - Now if force_permutation=True, and it fails to calculate permutation importance, the function raises an error (previous behavior skipped and tried model built-in importance, which is not intuitive and doesn't make sense IMO).

### CODE REVIEW

File: deepchecks/tabular/user_utils.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+def calculate_feature_importance(
+        model: t.Any,
+        dataset: t.Union['tabular.Dataset', pd.DataFrame],
+        permutation_kwargs: t.Dict[str, t.Any] = None,
+) -> pd.Series:
+    permutation_kwargs = permutation_kwargs or {}
+    permutation_kwargs['skip_messages'] = permutation_kwargs.get('skip_messages', True)
+    permutation_kwargs['timeout'] = permutation_kwargs.get('timeout', None)
+
+    if isinstance(dataset, pd.DataFrame):
```
Comment (Nadav-Barak) [inline deepchecks/tabular/user_utils.py:34]: I this is the case, why are you allowing dataset to be t.Union['tabular.Dataset', pd.DataFrame]?

File: deepchecks/tabular/user_utils.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+__all__ = [
+    'calculate_feature_importance'
+]
+
+
+def calculate_feature_importance(
+        model: t.Any,
+        dataset: t.Union['tabular.Dataset', pd.DataFrame],
+        permutation_kwargs: t.Dict[str, t.Any] = None,
```
Comment (Nadav-Barak) [inline deepchecks/tabular/user_utils.py:28]: I really dont like the permutation_kwargs for external methods, consider accepting the different parameters separately.

File: tests/tabular/test_user_utils.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+#
+"""Test user utils"""
+from hamcrest import assert_that, close_to, raises, calling
+from sklearn.neural_network import MLPClassifier
+
+from deepchecks.core.errors import DeepchecksValueError
+from deepchecks.tabular.user_utils import calculate_feature_importance
+
+
+def test_calculate_importance_when_no_builtin(iris_split_dataset_and_model):
```
Comment (Nadav-Barak) [inline tests/tabular/test_user_utils.py:19]: Add a test were you supply acceptable kwargs

File: deepchecks/tabular/user_utils.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+) -> pd.Series:
+    permutation_kwargs = permutation_kwargs or {}
```
Comment (Nadav-Barak) [inline deepchecks/tabular/user_utils.py:30]: doc string

**Comment by JKL98ISR** [comment]
can you give a better name for the pr?

File: deepchecks/utils/features.py [hunk 5]
```diff
@@ -150,15 +150,12 @@ def calculate_feature_importance(
 
     if force_permutation:
         if isinstance(dataset, pd.DataFrame):
-            permutation_failure = 'Cannot calculate permutation feature importance on a pandas Dataframe, using ' \
```
Comment (JKL98ISR) [inline deepchecks/utils/features.py:153]: why remove support for df?

Comment (nirhutnik) [reply deepchecks/utils/features.py:153]: It's removing the behavior that if you mark "force_permutation" then it later looks for built-in feature importance. Instead, it fails.

**Review by JKL98ISR — APPROVED** [review]

File: tests/tabular/test_feature_importance_user_utils.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+# ----------------------------------------------------------------------------
+#
+"""Test user utils"""
+from hamcrest import assert_that, close_to, raises, calling
+
+from deepchecks.core.errors import DeepchecksValueError
+from deepchecks.tabular.feature_importance import calculate_feature_importance
+
+
+def test_calculate_importance(iris_split_dataset_and_model):
```
Comment (Nadav-Barak) [inline tests/tabular/test_feature_importance_user_utils.py:18]: Minor and optional: Maybe in one of the test try using an alternative scorer

**Review by Nadav-Barak — APPROVED** [review]
Looks good!
Additional suggestion: maybe show case this option in some places in docs (specifically plots that calculate FI multiple times), see comment in board issue

**Review by ItayGabbay — APPROVED** [review]

**Pull request merged**